python/lsst/cp/pipe/deferredCharge.py

```diff
@@ -0,0 +1,1019 @@
+            ``"LAST_MEAN"``
+                Mean value of last image column (`float`).
+            ``"IMAGE_MEAN"``
+                Mean value of the entire image region (`float`).
+            ``"OVERSCAN_COLUMNS"``
+                List of overscan column indicies (`list` [`int`]).
+            ``"OVERSCAN_VALUES"``
+                List of overscan column means (`list` [`float`]).
+        calib : `lsst.ip.isr.DeferredChargeCalib`
+            Calibration to populate with values.
+        detector : `lsst.afw.cameraGeom.Detector`
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.DeferredChargeCalib`
+            Populated calibration.
+        """
+        # Range to fit.  These are in "camera" coordinates, and so
+        # need to have the count for last image column removed.
+        start, stop = self.config.globalCtiColumnRange
+        start -= 1
+        stop -= 1
+
+        # Loop over amps/inputs, fitting those columns from
+        # "non-saturated" inputs.
+        for amp in detector.getAmplifiers():
+            ampName = amp.getName()
+
+            # Number of serial shifts.
+            nCols = amp.getRawDataBBox().getWidth() + amp.getRawSerialPrescanBBox().getWidth()
+
+            # The signal is the mean intensity of each input, and the
+            # data are the overscan columns to fit.  For detectors
+            # with non-zero CTI, the charge from the imaging region
+            # leaks into the overscan region.
+            signal = []
+            data = []
+            for exposureEntry in inputMeasurements:
+                exposureDict = exposureEntry['CTI']
+                if exposureDict[ampName]['IMAGE_MEAN'] < self.config.maxSignalForCti:
+                    signal.append(exposureDict[ampName]['IMAGE_MEAN'])
+                    data.append(exposureDict[ampName]['OVERSCAN_VALUES'][start:stop+1])
+            signal = np.array(signal)
+            data = np.array(data)
+
+            ind = signal.argsort()
+            signal = signal[ind]
+            data = data[ind]
+
+            # CTI test.  This looks at the charge that has leaked into
+            # the first few columns of the overscan.
+            overscan1 = data[:, 0]
+            overscan2 = data[:, 1]
+            test = (np.array(overscan1) + np.array(overscan2))/(nCols*np.array(signal))
+            testResult = np.median(test) > 5.E-6
+            self.debugView(ampName, signal, test)
+
+            params = Parameters()
+            params.add('ctiexp', value=-6, min=-7, max=-5, vary=True)
+            params.add('trapsize', value=5.0 if testResult else 0.0, min=0.0, max=30.,
+                       vary=True if testResult else False)
+            params.add('scaling', value=0.08, min=0.0, max=1.0,
+                       vary=True if testResult else False)
+            params.add('emissiontime', value=0.35, min=0.1, max=1.0,
+                       vary=True if testResult else False)
```

Comment:
I think I agree that using the fixed cut-off isn't ideal.  The value chosen is the nominal "allowed serial CTI".  Therefore, this indicates when the measured CTI is above the threshold, the fit should continue to attempt to develop a correction to fix it.  Otherwise, the amplifier is within spec, and we can skip the fit.

Working through some toy models, and then reading through the paper some more, shows that figure 2 presents the kind of flux dependence you bring up.  At the lowest flux, there are some false positives, but those quickly drop below the threshold as the flux increases.  For the amplifiers that do have significant CTI effects, the test stays positive over much of the flux range.

Reply:
OK, thanks for the comments. What do you think is best to do here then? I think at a minimum we should have log messages announcing the behaviour though.

Reply:
I've added a log message that notes what is happening for each amplifier